app/models/application.rb

```diff
@@ -1,14 +1,41 @@
 require "securerandom"
 
 class Application < ApplicationRecord
-  attr_accessor :sponsor_types, :family_or_single_types, :living_space_types, :sponsor_type, :family_or_single_type, :living_space_type
+  POSTCODE_REGEX = /([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))\s?[0-9][A-Za-z]{2})/
+  MIN_PHONE_DIGITS = 9
+  MAX_PHONE_DIGITS = 14
+
+  attr_accessor :sponsor_types, :family_or_single_types, :living_space_types, :mobility_impairments_types, :accommodation_length_types, :dbs_certificate_types, :answer_more_questions_types,
+                :sponsor_type, :family_or_single_type, :living_space_type, :mobility_impairments_type, :accommodation_length_type, :dbs_certificate_type, :answer_more_questions_type, :single_room_count, :double_room_count, :postcode, :full_name, :email_address, :mobile_number
```

Comment:
email and fullname are already attributes and don't need to go into the json so we don't need to add an attr_accessor for them, Rails does this automagically